src/tests/JIT/opt/Devirtualization/Comparer_get_Default.cs

```diff
@@ -179,15 +231,66 @@ public static int TestEntryPoint()
                 var enumByteB = Unsafe.As<long, Enum_byte>(ref b);
                 Compare_Enum_Byte(enumByteA, enumByteB);
 
-                var structA = new Struct1 {a = a, b = b};
-                var structB = new Struct1 {a = b, b = a};
-                Compare_Struct1(structA, structB);
-
                 Compare_DateTime(
                     new DateTime(Math.Clamp(a, DateTime.MinValue.Ticks, DateTime.MaxValue.Ticks)),
                     new DateTime(Math.Clamp(b, DateTime.MinValue.Ticks, DateTime.MaxValue.Ticks)));
 
                 Compare_Int32_Nullable(a, b);
+                Compare_Enum_Int32_Nullable(enumIntA, enumIntB);
+
+                Compare_Int32_Nullable(null, b);
+                Compare_Enum_Int32_Nullable(null, enumIntB);
+
+                Compare_Int32_Nullable(a, null);
+                Compare_Enum_Int32_Nullable(enumIntA, null);
+
+                var structA = new Struct1 {a = a, b = b};
+                var structB = new Struct1 {a = b, b = a};
+                Compare_Struct1(structA, structB);
+
+                var struct2A = new Struct2 {a = a, b = b};
+                var struct2B = new Struct2 {a = b, b = a};
+                Compare_Struct2(struct2A, struct2B);
+
+                Compare_Struct1_Nullable(structA, structB);
+                Compare_Struct2_Nullable(struct2A, struct2B);
+
+                Compare_Struct1_Nullable(null, structB);
+                Compare_Struct2_Nullable(null, struct2B);
+
+                Compare_Struct1_Nullable(structA, null);
+                Compare_Struct2_Nullable(struct2A, null);
+
+                Compare_Struct1_Nullable(null, null);
+                Compare_Struct2_Nullable(null, null);
+
+                // workaround for: https://github.com/dotnet/roslyn/issues/68770
+                static T Bitcast<T>(long l) => Unsafe.As<long, T>(ref l);
```

Comment:
Would it be cleaned to just use standard `Unsafe.BitCast`?

Reply:
This casts to smaller types like `FloatEnum` too, `Unsafe.BitCast` would throw for those.